### DISCUSSION TITLE
Loading 'suite' from a Module

### DISCUSSION THREAD

**Discussion by Fenikkusu** [Codeception/Codeception#6718] [Q&A]
I have a suite that I drop into almost every project I use. While I do have a few changes I have to make from project to project, the suite is easily over 95% identical between projects. I'd like to ultimately convert this suite to a module that I include in any project that needs it, and use the Codeception config to handle the adjustments I am currently manually doing. I would want it to still run as an individual suite within the testing, with the ability to do `codecept run suite-name` and similar commands.

Has anyone done something of this nature?

**Comment by Naktibalda** [comment]
Does using git submodule count?
https://github.com/Naktibalda/codeception-mezzio-tests/tree/master/tests/functional

You could using `include` setting to execute tests from `vendor/` directory.
https://codeception.com/docs/Customization#One-Runner-for-Multiple-Applications

**Reply by Fenikkusu** [reply]
You're not wrong in that would likely work, though it seems a bit of a roundabout way of doing it.

**Reply by Naktibalda** [reply]
You can find multiple examples of people using `include` to run tests from `vendor/` in issues.
It is not a roundabout way - it was built for this purpose.

Regarding your idea.

Codeception modules can't load tests - they only provide action methods and some before/after hooks.
You could implement custom test loader as documented in https://codeception.com/docs/Customization#formats
but you would still need a suite.yaml file, suite directory and one file in that directory to trigger the loader.
And you would still have to figure out a way to load tests from your module.
